crates/ln-dlc-node/src/node/wallet.rs

```diff
@@ -104,4 +108,57 @@ impl Node {
             pending_close,
         }
     }
+
+    pub fn get_on_chain_history(&self) -> Result<Vec<bdk::TransactionDetails>> {
+        self.wallet
+            .on_chain_transactions()
+            .context("Failed to retrieve on-chain transaction history")
+    }
+
+    pub fn get_off_chain_history(&self) -> Vec<PaymentDetails> {
+        let inbound_payments = self
+            .inbound_payments
+            .lock()
+            .expect("to be able to acquire lock");
```

Comment:
Let's try to avoid expects from now on if possible

Reply:
I think `unwrap`ping a mutex lock is the normal way to handle this stuff. If a mutex is poisoned there is not much else to do and crashing the application is best. Here's [some context](https://www.reddit.com/r/rust/comments/10f6328/is_this_a_reasonable_way_to_handle_unwrapping/).

Reply:
Obviously a thread panicking and the rest of the application running as if nothing bad has happened is not ideal either. We could try something like [this](https://stackoverflow.com/a/74854542/6185443).

Eventually we should probably go down the route of supervising any long-running spawned threads and respawn them in case of panic, like we used to do in ItchySats with supervised actors.

Reply:
Fair enough :)

Reply:
another way of dealing with poisoned mutex is to evaluate whether we could ignore the poisoning...

https://users.rust-lang.org/t/mutex-poisoning-why-and-how-to-recover/72192/4
https://stackoverflow.com/questions/68426260/what-is-the-easiest-way-to-ignore-a-poison-error-on-mutex